diff --git a/modules/libmicroros/libmicroros.mk b/modules/libmicroros/libmicroros.mk
--- a/modules/libmicroros/libmicroros.mk
+++ b/modules/libmicroros/libmicroros.mk
@@ -38,7 +38,7 @@ get_package_names: $(COMPONENT_PATH)/micro_ros_src/src
 	@cd $(COMPONENT_PATH)/micro_ros_src/src; \
 	colcon list | awk '{print $$1}' | awk -v d=";" '{s=(NR==1?s:s d)$$0}END{print s}'
 
-configure_colcon_meta: $(COMPONENT_PATH)/colcon.meta $(COMPONENT_PATH)/micro_ros_src/src
+configure_colcon_meta: $(COMPONENT_PATH)/colcon.meta $(COMPONENT_PATH)/micro_ros_src/src $(ZEPHYR_CONF_FILE)
 	. $(COMPONENT_PATH)/utils.sh; \
 	cp $(COMPONENT_PATH)/colcon.meta $(COMPONENT_PATH)/configured_colcon.meta; \
 	ZEPHYR_CONF_FILE=$(ZEPHYR_CONF_FILE); \
